app/models/user.rb

```diff
@@ -73,9 +73,27 @@ def filter_accessible_app_groups(app_groups, roles: nil)
       role: (AppGroupRole.where(name: roles).pluck(:id) if roles)
     }.compact
 
+    where_clause_team = {}
+
+    group_ids = AppGroupTeam.where(app_group_id: app_groups.first.id).pluck(:group_id)
```

Comment:
So this function will be called from [here](https://github.com/BaritoLog/BaritoMarket/blob/master/app/models/user.rb#L67) and from [here](https://github.com/BaritoLog/BaritoMarket/blob/master/app/policies/app_group_policy.rb#L3). From the first reference we can be sure of that only 1 app group would be passed to _filter_accessible_app_groups_ function. Since the output will be an array of only one app group everytime, I used _first_

Reply:
if so, can we make the argument to be an object instead of an array? because it is confusing

Reply:
Updated to have support for multiple app groups. My [previous](https://github.com/BaritoLog/BaritoMarket/pull/289#discussion_r1505475622) reference that only one app group will be present for every call of _filter_accessible_app_groups_ was wrong. There is a function [here](https://github.com/BaritoLog/BaritoMarket/blob/master/app/policies/app_group_policy.rb#L47) in scope which can have multiple app groups in one call